compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -580,8 +580,9 @@ private void findTypeParamInTupleForArray(Location loc, BArrayType expType, BTup
         if (actualType.restType != null) {
             tupleTypes.add(actualType.restType);
         }
-        BUnionType tupleElementType = BUnionType.create(null, tupleTypes);
-        findTypeParam(loc, expType.eType, tupleElementType, env, resolvedTypes, result);
+
+        BType type = tupleTypes.size() == 1 ? tupleTypes.iterator().next() : BUnionType.create(null, tupleTypes);
```

Comment:
It is not valid to create a union type when the size of the `tupleTypes` array is zero. Attempting to do so will cause the following program to crash. I have created a separate [issue](https://github.com/ballerina-platform/ballerina-lang/issues/39699) for this problem
```ballerina
import ballerina/io;

const mm = "s";
public function main() {
    var data = [].map( m => m + 1);
    io:println(data);
}
```